clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetcher.java

```diff
@@ -99,8 +99,8 @@
     private final ConcurrentLinkedQueue<CompletedFetch> completedFetches;
     private final Deserializer<K> keyDeserializer;
     private final Deserializer<V> valueDeserializer;
-
     private PartitionRecords nextInLineRecords = null;
+    private ExceptionMetadata nextInLineExceptionMetadata = null;
```

Comment:
If an `InvalidRecordException` is thrown in `Fetcher.fetchRecords()` the position of that partition won't advance and no message for that partition would be added to `fetched` either. So it seems we can apply the same rule there as well?